### ISSUE TITLE
Boolean field in smart contract cannot be set, always false.

### ISSUE THREAD

**Issue by TechUpDesigns** [greymass/anchor#1300]
### Description

It looks like this bug: https://github.com/greymass/anchor/issues/1250
Has resurfaced.

No matter what I set for a value in the boolean field of a smart contract, Anchor shows it as false. I have tried Windows Desktop wallet and the iOS mobile wallet.

I have tried  1 and 0 for true and false when setting up the action, both result in False in Anchor.
I have tried JavaScript True and False when setting up the action, both result in False in Anchor.
The same code in Scatter/TokenPocket recognizes the boolean field and sets the value to True/False correctly.

The screenshots below show Anchor not recognizing the boolean value, one screenshot shows TokenPocket recognizing the boolean value correctly.

![TokenPocket correctly recognizes value](https://user-images.githubusercontent.com/24238654/185758253-141afb3c-4322-4584-a126-a1bbf678a201.jpg)
![Mobile_confirmation](https://user-images.githubusercontent.com/24238654/185758255-1d17e4ff-18de-41c6-86b4-b53b3ab4c7b7.jpg)
![Mobile app](https://user-images.githubusercontent.com/24238654/185758256-965d6003-6ca6-4ae0-a188-c60244bc3e2b.jpg)
![windows desktop](https://user-images.githubusercontent.com/24238654/185758258-a131b307-87ae-42c2-ba2a-f5430e013c2d.jpg)
![windows confirmation screen](https://user-images.githubusercontent.com/24238654/185758259-a4e25e93-6d34-4787-bbca-26cea8a873b3.jpg)
![windows using latest version](https://user-images.githubusercontent.com/24238654/185758260-bb813dca-8efd-4167-b90f-7d52f49d79a8.jpg)

### Platform

All

### Steps To Reproduce

Visit: https://www.eoserc.com/
Login with Anchor Login.
Click Add New Account.
Toggle Enable.
Click Add.

The Anchor confirmation window will show enable as False.

### Contact Details

techupdesigns@gmail.com

**Comment by aaroncox** [comment]
This looks to be an issue with the signing request protocol. The request Anchor is receiving from the app indeed has a `enabled: false` value, regardless of which version of Anchor you use or even if you view the ESR payload manually. Here's the payload Anchor is receiving for the transaction, which you can also see shows that value as false:

https://eosio.to/gmNgZDrylDn5hNY8I43nDCIMDEwBmnMlpF9eTWUAASCHUWBqHFzA55mBK8OExSxcsb4bdxWIXgsvjAo2Ypxw9YroxU7fYpCKhtVqy5VgAjCalQECYDQQMFlnlJQUFFvp6ycn6SXmJWfkF-nlZOZl61smGRqnpBia6BqlWCbrmiQZmusmGhgZ6pomJaWZmqaYWloamzKxJucXZ6Y7MzLIM07TWSzNHxwZ83uDrOjNWKPmlgCPXye2PH3C8ObozJxV0iVvYoul1d-sOr676ON0N5c8ngOzcipPnC2p_FvQx3nv7BkWkKUs_MAwAAA

I think the issue is coming from something either in the app or in the SDKs in use, since I can manually craft an ESR payload that actually has the true value and that pops up in Anchor:

https://eosio.to/gmNgZGQK0JwrIf3yaioDCAA5jAJT4-ACPs8MXBkmLGbhivXduKtA9Fp4YVSwEeOEq1dEL3b6FoNUNKxWW64EE4DRrAwQAKMZGZisM0pKCoqt9PWTk_QS85Iz8ov0cjLzsvUNTI0tE5NNjXRTUyySdU3MzFN0E43NUnTN0pLNTS0skpMSDSwYAA

Is this application open source? Is there a way we can see the stack it's built upon, or can you list out which SDKs for EOSIO are in use? 

Something is causing the transaction to have a false value before it even reaches Anchor. It might be one of our SDKs or something else, not sure without more details.

**Comment by TechUpDesigns** [comment]
Hi Aaron,

Thank you for taking the time to look into this. I've invited you to our
repo.
Since then I've updated to the latest versions of anchor-link and
anchor-link-browser-transport.
Yet the issue is still occurring.
When the action is sent for signing it does not recognize the boolean
value.
The same action sent to Scatter/TokenPocket does recognize the boolean
value.

Could this be an issue with either of the JS libraries?

Thank you again for looking into this. I appreciate your time.
